pydm/widgets/slider.py

```diff
@@ -73,14 +80,157 @@ def __init__(self, parent=None, init_channel=None):
+        """
+        self.widget = QWidget()
+        self.widget.show()
+
+        self.widget.setWindowTitle("PyDM Slider Parameters")
+        self.widget.resize(300, 200)
+
+        main_layout = QVBoxLayout(self.widget)
+
+        self.input = []
+        self.label = []
+        self.button = []
+        self.menu_layout = []
+
+        text_info = ['Value', 'Increment', 'Increment scale', 'Precision', 'Precision from PV', 'Number Format',
+                     'OK', 'Apply', 'Cancel']
+
+        combo_box_table_scale = ['1e0', '1e1', '1e2', '1e3', '1e4', '1e5']
+        combo_box_table_format = ['Float', 'Exp']
+
+        for key in range(0, 6):
+            self.input.append(key)
+            self.label.append(key)
+            self.menu_layout.append(key)
+
+            self.menu_layout[key] = QHBoxLayout()
+            self.menu_layout[key].setAlignment(Qt.AlignLeft)
+
+            self.label[key] = PyDMLabel(self.widget)
+            self.label[key].setText(text_info[key])
+            self.menu_layout[key].addWidget(self.label[key])
+
+            if key == 4:
+                self.input[key] = QCheckBox()
+                self.input[key].setTristate(on=False)
+                self.input[key].setChecked(self.precisionFromPV)
+            elif key == 2 or key == 5:
+                self.input[key] = QComboBox()
+            else:
+                self.input[key] = QLineEdit()
+                self.input[key].setText("")
+
+            self.menu_layout[key].addWidget(self.input[key])
+            main_layout.addLayout(self.menu_layout[key])
+
+        self.menu_layout.append(3)
+        self.menu_layout[3] = QHBoxLayout()
+
+        self.input[0].setText(str(self.value))
+        self.input[1].setText(str(self._step_size))
+        self.input[3].setText(str(self.precision))
+
+        self.input[2].addItems(combo_box_table_scale)
+        self.input[5].addItems(combo_box_table_format)
+
+        for key in range(0, 3):
+            self.button.append(key)
+            self.button[key] = QPushButton(self.widget)
+            self.button[key].setText(text_info[key+6])
+            self.menu_layout[3].addWidget(self.button[key])
+
+        main_layout.addLayout(self.menu_layout[3])
+
+        self.button[0].clicked.connect(self.apply_and_close_menu)
+        self.button[1].clicked.connect(self.apply_step_size_menu_changes)
+        self.button[2].clicked.connect(self.widget.close)
+
+    def apply_and_close_menu(self):
+        """
+        Method for the 'ok' button in the slider parameters menu.
+        """
+        self.apply_step_size_menu_changes()
+        self.widget.close()
+
+    def apply_step_size_menu_changes(self):
+        """
+        Method which attempts to set the user imputed data from the slider parameters menu.
+        """
+        try:
+            val = float(self.input[1].text())
+            val = val*float(self.input[2].currentText())
+            if val > 0:
+                self.step_size = val
```

Comment:
A more descriptive name than val would help here (and perhaps later in this method as well). It's hard to tell what is actually being multiplied here without running the code and opening the menu.